### PR TITLE
Minor: Exchange cosmetics

### PR DESCRIPTION
**Pull request by hroff-1902**
* Exception message fixed (should also be done for FTX in/after #3438)
* Minor fix in a usecase
* Added validation for `price <= stop_price` (actually, `0.0 < stoploss_on_exchange_limit_ratio < 1.0`, which is equivalent) at the startup time. Now exception is raised surprisingly at the creation of first stoploss, which can happen in days.

For the last item, to be discussed:
* I left the duplicating check in exchange/binance.py, it can be removed, but this check makes the stoploss() method more error-resistant (maybe)
* The stoploss_on_exchange_limit_ratio can only be validated at startup when stoploss_on_exchange = True or even only for Binance (and for FTX after #3438) exchange only, i.e. when it's actually used.

Inspired by the new comment arrived: https://github.com/freqtrade/freqtrade/issues/2076#issuecomment-643663739

### CODE REVIEW

File: freqtrade/exchange/exchange.py [hunk 1]
```diff
@@ -386,6 +386,13 @@ def validate_ordertypes(self, order_types: Dict) -> None:
                 f'On exchange stoploss is not supported for {self.name}.'
             )
 
+        # Limit price threshold: As limit price should always be below stop-price
+        # Used for limit stoplosses on exchange
+        limit_price_pct = order_types.get('stoploss_on_exchange_limit_ratio', 0.99)
+        if limit_price_pct >= 1.0 or limit_price_pct <= 0.0:
+            raise OperationalException(
+                "stoploss_on_exchange_limit_ratio should be < 1.0 and > 0.0")
+
```
Comment (xmatthias) [inline freqtrade/exchange/exchange.py:395]: I don't think this is exchange specific - so mabye adding it to the validation in constants.py would suffice? (you can add min / max values there ...)

Comment (xmatthias) [reply freqtrade/exchange/exchange.py:397]: with this addition to constants, the following validation comes up:

```
                'stoploss_on_exchange_limit_ratio': {'type': 'number', 'minimum': 0.0,
                                                     'maximum': 1.0}
```

```
2020-07-13 07:14:22,436 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2020-07-13 07:14:22,437 - freqtrade.configuration.config_validation - CRITICAL - Invalid configuration. See config.json.example. Reason: -1.0 is less than the minimum of 0.0

Failed validating 'minimum' in schema['properties']['order_types']['properties']['stoploss_on_exchange_limit_ratio']:
    {'maximum': 1.0, 'minimum': 0.0, 'type': 'number'}

On instance['order_types']['stoploss_on_exchange_limit_ratio']:
    -1.0
2020-07-13 07:14:22,439 - freqtrade.commands.trade_commands - ERROR - -1.0 is less than the minimum of 0.0

```

This is followed by another, "not so nice" stacktrace - but i'd rather take that in a seperate PR.

@hroff-1902 if you agree, i'll commit this to this branch and remove the above, explicit test (as that cannot be hit anymore ...).

Comment (hroff-1902) [reply freqtrade/exchange/exchange.py:398]: yes, please, let's complete it...

I remember I was wanted to revert some code in the commits above, and cannot recall in the meantime how I wanted to complete this PR a month ago, following your comments above.

Comment (xmatthias) [reply freqtrade/exchange/exchange.py]: for sure - let me know what you think - it's the last (2) commits which remove this test in place of testing it as part of the configuration.

**Comment by hroff-1902** [comment]
thanks,
LGTM 👍

**Pull request merged**